### ISSUE TITLE
[Feature] Option to hide toolbar

### ISSUE THREAD

**Issue by chromer030** [Zettlr/Zettlr#2976]
### Checklist

- [X] Zettlr does not possess this feature (if it does, but the feature doesn't work out for you, propose an enhancement instead.)
- [X] Zettlr's current features cannot be used to approximate this behaviour (otherwise, propose an enhancement instead.)
- [X] I have searched the documentation to make sure I did not overlook it.
- [X] I have searched both open and closed issues to see if someone has already proposed this.

### Feature Proposal

Please provide an option to hide the toolbar.
Thank you.

### Example Workflow

This feature makes the app UI more minimal and give the users with small displays more space for work.

### Caveats

I think this feature doesn't impede other features.

### Do you wish to attempt implementing this yourself?

No

### Platforms

- [X] Windows
- [X] macOS
- [X] Linux

**Comment by nathanlesage** [comment]
Do you wish to hide the toolbar *always*? You should be able to hide it during distraction free in the current nightlies, I implemented that recently

**Comment by chromer030** [comment]
Thank you very much for considering this great features for this lovely app , please consider an option to hide it permanently also.

**Comment by chchen2021** [comment]
The CSS snippet to [Hide the Toolbar in Distraction Free](https://docs.zettlr.com/en/core/custom-css/#hide-the-toolbar-in-distraction-free) does not seem to work in Zettlr 2.1.0.

I tried the nightly build mentioned above. When the toolbar is "hidden", the same bar is still there but the title is simplified to "Zettlr". Can you tweak it so the title bar is complete gone. That way, the distraction-free mode truly offers a blank piece of paper.

**Comment by nathanlesage** [comment]
> The CSS snippet to Hide the Toolbar in Distraction Free does not seem to work in Zettlr 2.1.0.

That is correct, we need JavaScript to do so now. A toggle for hiding it during distraction free has been implemented and is coming with the next update.

> Can you tweak it so the title bar is completely gone? 

That "same bar" is actually just a title bar, because that is what you will still need. Otherwise, the window becomes un-draggable which is certainly not what you want, is it …?

**Comment by chchen2021** [comment]
I see. Native macOS apps automatically hide the title bar in full-screen mode, but reveals it when the mouse moves over the bar. If the title bar remains, I actually prefer the default toolbar, which occupies the same space but is prettier and more informative.

**Comment by nathanlesage** [comment]
> in full-screen mode

Aaaaah. Yes. I don't know, we are able to detect if we're in Fullscreen or not, so we should be able to hide the toolbar/titlebar completely there. BUT I don't think we can detect if the user is hovering over the menu bar as to pull down this toolbar.

Electron is, after all, still not fully native, it's just a fad!

**Comment by chchen2021** [comment]
I just tested on macOS Monterey, the mouse-over reveal of the title bar **and** the menu bar applies to Zettlr too. So I think it is safe to completely hide the toolbar in full-screen mode. In the full-screen mode, there is no need to drag any window.

**Comment by nathanlesage** [comment]
Ah right, I see. Feel free to open another issue. Making this possible requires some logical magic and thus might take a little bit of time, but I see the value.

**Comment by chchen2021** [comment]
I will be happy to open another issue, but I want to understand why you need a logic here. If you only offer to hide the toolbar in the full-screen mode, there is no need for a logic. 

In the normal mode, you can keep the toolbar as is. I have already tested the current nightly build, and suspect that most users do not need this intermediate "hiding" while the title bar persists with just a different title -- the big "Zettlr" title is really not as pretty as the smaller icons, and is not offering any clickable function either.

**Comment by nathanlesage** [comment]
> why you need a logic here

This is the current logic:

```typescript
shouldShowToolbar: function (): boolean {
  return this.distractionFree === false || this.$store.state.config['display.hideToolbarInDistractionFree'] === false
},
shouldShowTitlebar: function (): boolean {
  return this.shouldShowToolbar === false
},
```

If we also have to account for fullscreen mode (including a check if we're running macOS, because on Windows and Linux this will basically make the app un-closable), this means that this logic will be even more important. I simply want to make sure I don't forget one edge case. I think on Windows & Linux the titlebar is shown regardless of any setting, but that is why I want to focus on this while I have some free time and can think straight so I don't make an error :)

**Comment by chromer030** [comment]
We love this feature !

**Comment by danielkarelpetr** [comment]
This works:

#editor.fullscreen {
  top:0;  
  position:fixed;  
}

**Comment by kromsam** [comment]
> This works:
> 
> #editor.fullscreen { top:0; position:fixed; }

this one does not work (anymore?)

**Comment by doctorcolossus** [comment]
> > This works:
> > #editor.fullscreen { top:0; position:fixed; }
> 
> this one does not work (anymore?)

Yeah, it doesn't work for me either... Has anyone got another solution?

I would personally like to hide it almost always... I don't really need or want any of the other stuff from "distraction-free mode". The only thing I find distracting so far is the toolbar, which I don't want taking up space on my screen. I prefer to learn keyboard shortcuts or use dropdown menus for actions whose shortcuts I haven't learned yet.

It would be ideal to have a "toggle toolbar" option under the "view" dropdown menu akin to the existing "toggle file manager" and "toggle sidebar" options, with an accompanying keyboard shortcut.

**Comment by danielkarelpetr** [comment]
Hi, try this:

.main-editor-wrapper.fullscreen .ͼ1 .cm-scroller {  
    top:0;
    position:fixed;   
}

.main-editor-wrapper.fullscreen .ͼ10 .cm-scroller {  
    top:0;
    position:fixed;    
}

**Comment by doctorcolossus** [comment]
Thanks, but that doesn't seem to work for me either!
<img width="2560" height="1440" alt="Image" src="https://github.com/user-attachments/assets/65e9346a-2049-456b-8bb4-fde1aca7815c" />
<img width="2560" height="1440" alt="Image" src="https://github.com/user-attachments/assets/9bf79b0d-0cbc-47fe-82f7-98e750e791be" />

I made sure to save, and also tried restarting.

**Comment by danielkarelpetr** [comment]
This works in distraction free mode (CTRL+J).

**Comment by doctorcolossus** [comment]
Maybe for you, but not for me. Anyway, as stated in my original comment, I do not want "distraction-free mode" and its other opinionated changes. I want "normal" mode without the toolbar, or to be able to toggle the toolbar regardless of the mode. For now I have switched to [marktext](https://github.com/marktext/marktext), which has no toolbar.

<img width="2560" height="1440" alt="Image" src="https://github.com/user-attachments/assets/cb9ba93d-8bf5-4df8-9c87-61a3b84a5488" />

**Comment by danielkarelpetr** [comment]
I was just trying to help. I don't actually use Zettlr at the moment.

My original comment was specifically about Distraction Free mode, so I simply updated the CSS to work with the latest version of Zettlr. Also, keep in mind that this works on Windows. If you're using Linux, the CSS may need to be adjusted.

**Comment by doctorcolossus** [comment]
> I was just trying to help.

I understand, and I do appreciate that.

> I don't actually use Zettlr at the moment.

I've also found an alternative I'm relatively happy with, as mentioned in my last comment, but I commented here before searching for alternatives and was also following up to be polite and possibly help test for others who might experience the same frustration in the future.

> Also, keep in mind that this works on Windows. If you're using Linux, the CSS may need to be adjusted.

Perhaps that's it, but it seems strange that the CSS would differ based on OS.

In any case, thanks for your suggestion!
